### DISCUSSION TITLE
When I ran a command with timeit, I am getting the time taken information but not the command output

### DISCUSSION THREAD

**Discussion by emailstorbala** [nushell/nushell#12031] [Q&A]
I am trying to run a command in nushell (0.90.1) with `timeit` command. I want to get both the time taken by the command as well as the command output. Is there anyway to do this?

![image](https://github.com/nushell/nushell/assets/12377900/8235028c-2c94-4a29-9f4e-17c4652fa504)

**Comment by fdncred** [comment]
That is working as intended.

**Reply by emailstorbala** [reply]
@fdncred Does it not work for the in-built command alone?

**Reply by fdncred** [reply]
i'm not sure what you mean. you're showing it working for `ls` in your screenshot, which is built-in.

**Reply by emailstorbala** [reply]
@fdncred In my screenshot, I have run the command `timeit` against the built-in `ls` command. In that, I see only the `time` information but not the command output.
Hence was asking whether the `timeit` command doesn't work with the built-in command to provide both `time` details and `command output` details ?

**Reply by ghost** [reply]
it is intended for the output of ls to be removed when running it within timeit.

and im currently not able to see why you would want both at once:

- when comparing ways to do something you don't want stuff (like compiler-logs) in between
- when just checking the speed of something you usually don't need the output
- when just running things it would be counterproductive since it would prevent proper parsing of anything
- it would also prevent easy parsing of timeit output

the one case where both is interesting for me is if i want to be informed "yes that was slow - 12sec to be exact" during normal interactive sessions when git was slow or something.  
and for that you build that functionality into your prompt where it does not interfere with anything (if you dont want to do it yourself: starship, etc have it builtin)


if you really need both you could probably use something like this:
```nu
def timeit_with_output [closure] {
  mut output = null
  let time = (timeit {$output = (do $closure)})
  {
    output: $output
    time: $time
  }
}
```
```

**Reply by emailstorbala** [reply]
@Jan9103 Thank you for the explanation!
Actually I don't want to `timeit` any built-in command like `ls`. But in my usual day-to-day activities, I use some product deployment scripts with terraform (called from python -> bash script -> terraform). I had a bash script to automate the arguments passed on to this python script.

Currently I migrated this `bash` script to `nushell`. While doing so, I replaced the existing `time` command in the bash script with `timeit` command like below.
`timeit run-external ./deployprod.py ...$args`
But due  to some weird unknown reasons, I am getting the output of the deployment command properly but not the time taken for the command. Hence I was trying to analyse the issue with some bizarre commands. Sorry.

**Reply by ldoguin** [reply]
What about running long import tasks that might output successful and failed operations, possibly with info to retry failed operations ? Granted a task like this could include the time it took. But I am assuming it's much easier to wrap it in a timeit in many cases.
